File: src/test/java/net/flyingfishflash/loremlist/unit/domain/lrmitem/LrmItemNotFoundTests.java
package net.flyingfishflash.loremlist.unit.domain.lrmitem;

import static org.assertj.core.api.Assertions.assertThat;

import java.util.Set;
import java.util.UUID;
import net.flyingfishflash.loremlist.domain.lrmitem.ItemNotFoundException;
import org.junit.jupiter.api.Test;

class LrmItemNotFoundTests {

  @Test
  void primaryConstructorDefaultMessage() {
    UUID id = UUID.randomUUID();
    ItemNotFoundException exception = new ItemNotFoundException(id);
    assertThat(exception.getMessage()).isEqualTo("Item could not be found.");
  }

  @Test
  void primaryConstructorNonDefaultMessage() {
    ItemNotFoundException exception = new ItemNotFoundException(UUID.randomUUID(), "Lorem Ipsum");
    assertThat(exception.getMessage()).isEqualTo("Lorem Ipsum");
  }

  @Test
  void secondaryConstructorDefaultMessageSetSizeOne() {
    Set<UUID> ids = Set.of(UUID.randomUUID());
    ItemNotFoundException exception = new ItemNotFoundException(ids);
    assertThat(exception.getMessage()).isEqualTo("Item could not be found.");
  }

  @Test
  void secondaryConstructorDefaultMessageSetSizeGreaterThanOne() {
    Set<UUID> ids = Set.of(UUID.randomUUID(), UUID.randomUUID());
    ItemNotFoundException exception = new ItemNotFoundException(ids);
    assertThat(exception.getMessage()).isEqualTo("Items (2) could not be found.");
  }

  @Test
  void secondaryConstructorNonDefaultMessage() {
    ItemNotFoundException exception =
        new ItemNotFoundException(Set.of(UUID.randomUUID()), "Lorem Ipsum");
    assertThat(exception.getMessage()).isEqualTo("Lorem Ipsum");
  }
}

File: src/test/kotlin/net/flyingfishflash/loremlist/unit/domain/lrmitem/LrmItemNotFoundTests.kt
package net.flyingfishflash.loremlist.unit.domain.lrmitem

import io.kotest.core.spec.style.DescribeSpec
import io.kotest.matchers.shouldBe
import net.flyingfishflash.loremlist.domain.lrmitem.ItemNotFoundException
import java.util.*

class LrmItemNotFoundTests :
  DescribeSpec({
    describe("ItemNotFoundException()") {
      it("primary constructor: default message") {
        val id = UUID.randomUUID()
        val itemNotFoundException = ItemNotFoundException(id)
        itemNotFoundException.message.shouldBe("Item could not be found.")
      }

      it("primary constructor: non-default message") {
        val itemNotFoundException = ItemNotFoundException(UUID.randomUUID(), "Lorem Ipsum")
        itemNotFoundException.message.shouldBe("Lorem Ipsum")
      }

      it("secondary constructor: default message, set size = 1") {
        val ids = setOf(UUID.randomUUID())
        val itemNotFoundException = ItemNotFoundException(ids)
        itemNotFoundException.message.shouldBe("Item could not be found.")
      }

      it("secondary constructor: default message, set size > 1") {
        val ids = setOf(UUID.randomUUID(), UUID.randomUUID())
        val itemNotFoundException = ItemNotFoundException(ids)
        itemNotFoundException.message.shouldBe("Items (2) could not be found.")
      }

      it("secondary constructor: non-default message") {
        val itemNotFoundException = ItemNotFoundException(setOf(UUID.randomUUID()), "Lorem Ipsum")
        itemNotFoundException.message.shouldBe("Lorem Ipsum")
      }
    }
  })
